plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiSmokeTest.java

```diff
@@ -165,6 +167,31 @@ public void testPartitionColumn()
         assertQueryFails("SELECT \"$partition\" FROM " + HUDI_NON_PART_COW, ".* Column '\\$partition' cannot be resolved");
     }
 
+    @DataProvider(name = "timeTravelQueryProvider")
+    public Object[][] timeTravelQueryProvider()
+    {
+        return new Object[][] {
+                {"SELECT count(_hoodie_commit_time) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-16 07:14:53.74700 Z'", "VALUES 99"},
+                {"SELECT count(1) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-16 07:14:53.74700 Z'", "VALUES 99"},
+                {"SELECT count(*) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-16 07:14:53.74700 Z'", "VALUES 99"},
+                {"SELECT count(_hoodie_commit_time) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216071453747", "VALUES 99"},
+                {"SELECT count(1) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216071453747", "VALUES 99"},
+                {"SELECT count(*) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216071453747", "VALUES 99"},
+                {"SELECT count(_hoodie_commit_time) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216009999999", "VALUES 0"},
+                {"SELECT count(1) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216009999999", "VALUES 0"},
+                {"SELECT count(*) FROM STOCK_TICKS_COW FOR VERSION AS OF 20211216009999999", "VALUES 0"},
+                {"SELECT count(_hoodie_commit_time) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-15 16:00:00.000000000 Z'", "VALUES 0"},
+                {"SELECT count(1) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-15 16:00:00.000000000 Z'", "VALUES 0"},
+                {"SELECT count(*) FROM STOCK_TICKS_COW FOR TIMESTAMP AS OF TIMESTAMP '2021-12-15 16:00:00.000000000 Z'", "VALUES 0"},
+        };
+    }
+
+    @Test(dataProvider = "timeTravelQueryProvider")
+    public void testTimeTravelQuery(@Language("SQL") String sql, String expected)
+    {
+        assertQuery(sql, expected);
+    }
```

Comment:
Could you add a test case for schema evolution in Hudi (e.g. add / drop column)?

Reply:
Hudi connector have not supported add/ drop column yet.
I am working on the schema revolution for hudi query now(plan to submit PR 2 or 3 weeks later).
Is possible to add a test case for schema evolution in the schema revolution PR?

Reply:
I mean schema evolution on Spark Hudi.

Reply:
got you, i will update

Reply:
added one schema evolution case on Spark Hudi.